thorlcr/slave/thslavemain.cpp

```diff
@@ -195,7 +198,9 @@ void UnregisterSelf(IException *e)
 bool ControlHandler(ahType type)
 {
     if (ahInterrupt == type)
-        LOG(MCdebugProgress, thorJob, "CTRL-C pressed");
+        LOG(MCdebugProgress, thorJob, "CTRL-C caught");
+    if (ahTerminate == type)
+        terminated = true;
```

Comment:
The signal may not originate from the thor script on a clean shutdown.
In which case Thor is still running and this is going to cause slaves to exit, without unregistering from the master.